src/elements/Header/_Header.js

```diff
@@ -1,42 +1,98 @@
 import cx from 'classnames'
-import React, { createElement, Component, PropTypes } from 'react'
-
+import React, { PropTypes } from 'react'
 import META from '../../utils/Meta'
+import {
+  getUnhandledProps,
+  iconPropRenderer,
+  imagePropRenderer,
+  useKeyOnly,
+} from '../../utils/propUtils'
+
+function _Header(props) {
+  const {
+    _sdClass, _headerElement,
+    left, center, right, justified, dividing, block, attached, floating,
+    icon, image, children, className,
+  } = props
+
+  const classes = cx(
+    _sdClass, 'ui',
+    useKeyOnly(left, 'left aligned'),
+    useKeyOnly(center, 'center aligned'),
+    useKeyOnly(right, 'right aligned'),
+    useKeyOnly(justified, 'justified'),
+    useKeyOnly(dividing, 'dividing'),
+    useKeyOnly(block, 'block'),
+    useKeyOnly(attached, 'attached'),
```

Comment:
Label seems to use `useValueAndKey(attached, 'attached'),` rather than 
what you specify below.  Is there reason for this?
> Levi Thomason <mailto:notifications@github.com>
> 24 June 2016 at 17:50
>
> In src/elements/Header/_Header.js 
> <https://github.com/TechnologyAdvice/stardust/pull/278#discussion_r68426699>:
>
> > +function _Header(props) {
> > + const {
> > + _sdClass, _headerElement,
> > + left, center, right, justified, dividing, block, attached, floating,
> > + icon, image, children, className,
> > + } = props
> > +
> > + const classes = cx(
> > + _sdClass, 'ui',
> > + useKeyOnly(left, 'left aligned'),
> > + useKeyOnly(center, 'center aligned'),
> > + useKeyOnly(right, 'right aligned'),
> > + useKeyOnly(justified, 'justified'),
> > + useKeyOnly(dividing, 'dividing'),
> > + useKeyOnly(block, 'block'),
> > + useKeyOnly(attached, 'attached'),
>
> The attached <http://semantic-ui.com/elements/header#attached> option 
> appears as |top attached|, |attached|, and |bottom attached| html 
> classes. In this case the key /or/ the value and key are used. They 
> are express in the API as:
>
> <Header attached />
> <Header attached='top' />
> <Header attached='bottom' />
>
> Then in the className buildup as:
>
> cx(
> useKeyOrValueAndKey(attached, 'attached'),
> )
>
> This method will use the boolean value (attached) as the className if 
> no value was provided. It will also handle the value if provided. 
> Hence the awkward but accurate naming "use key /or/ value and key".
>
> ------------------------------------------------------------------------
>
> For comparison, the current implementation would always produce this 
> for all the above API permutations:
>
> <div class="ui attached header"></div>
>
> Because only the key "attached" is being used. Use |useKeyOnly| for 
> boolean style classes, classes that can appear in tandem with other 
> classes, are either present or not, and do not have any combination 
> classes.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/TechnologyAdvice/stardust/pull/278/files/4317290b6fd2e4980cfd74e736287438f69d4c6c#r68426699>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AAWHsxeoP7PeGInj9OAf7QvegYpGi44zks5qPArwgaJpZM4I9saB>.
>

-- 
Sent from Postbox 
<https://www.postbox-inc.com/?utm_source=email&utm_medium=siglink&utm_campaign=reach>